lua/nvim-treesitter/utils.lua

```diff
@@ -193,4 +193,13 @@ function M.to_func(a)
   return type(a) == "function" and a or M.constant(a)
 end
 
+function M.ts_cli_version()
+  if fn.executable "tree-sitter" == 1 then
+    local handle = io.popen "tree-sitter  -V"
+    local result = handle:read "*a"
+    handle:close()
+    return vim.split(result, "\n")[1]:match "[^tree%psitter ].*"
```

Comment:
Wo this regex looks weird, are you sure you wanted that?

Reply:
I stole this 1:1 from the healthcheck function ;)

Reply:
So the point here is that `tree-sitter -V` outputs something like
```
tree-sitter 0.20.3 (<commit hash if built from source>)
```
and we need to extract the number for comparison.

Reply:
But why adding the `[]` that's what I don't get

Reply:
🤷🏻 

Lua regex is still a bit of a mystery to me; I just know that this works for our purpose.

Reply:
I should not say "regex"; Lua does _not_ have regular expressions, it just has pattern matching. In this case (quoting from the [Lua-users wiki](http://lua-users.org/wiki/PatternsTutorial):

> You can also create your own classes by wrapping a group of characters in square brackets. This will match one of the characters. If the first character inside the brackets is ^, then it will match a character not in the group.